### PR TITLE
res_pjsip: Increase pj_sip realm size

### PR DESCRIPTION
**Pull request by pchero**
res_pjsip: Enlarge pj_sip auth realm size to 255 Characters

This commit extends the realm size for pj_sip auth from its 
existing 40-character limitation to a more versatile 255-character 
size. 
This enhancement addresses limitations in domain qualification 
and practical use, resulting in improved functionality. 
Additionally, it includes changes to the realtime database schema 
and ARI message length to accommodate the increased realm size.

Resolves: #345

### CODE REVIEW

File: include/asterisk/res_pjsip.h [hunk 1]
```diff
@@ -87,6 +87,8 @@
 #define AST_STIR_SHAKEN_RESPONSE_STR_UNSUPPORTED_CREDENTIAL "Unsupported Credential"
 #define AST_STIR_SHAKEN_RESPONSE_STR_INVALID_IDENTITY_HEADER "Invalid Identity Header"
 
+#define PJSIP_AUTH_MAX_REALM_LENGTH 255	/* From the auth/realm realtime column size */
```
Comment (pchero) [inline include/asterisk/res_pjsip.h:90]: Introduced a new definition for realm size.

Comment (seanbright) [inline include/asterisk/res_pjsip.h:90]: We should try to avoid prefixing things with `PJSIP_` in case the upstream project wanted to use it, `AST_SIP_AUTH_MAX_REALM_LENGTH` instead maybe?

File: include/asterisk/res_pjsip.h [hunk 2]
```diff
@@ -3529,7 +3531,7 @@ struct ast_sip_transport_state *ast_sip_get_transport_state(const char *transpor
 
 /*!
  * \brief Return the SIP URI of the Contact header
- * 
+ *
```
Comment (seanbright) [inline include/asterisk/res_pjsip.h:3534]: Minor: unnecessary whitespace change here.

File: res/ari/internal.h [hunk 3]
```diff
@@ -59,7 +59,7 @@ struct ast_ari_conf {
 };
 
 /*! Max length for auth_realm field */
-#define ARI_AUTH_REALM_LEN 80
+#define ARI_AUTH_REALM_LEN 256
```
Comment (seanbright) [inline res/ari/internal.h:62]: This is for ARI and has nothing to do with SIP/PJSIP. If you still want to change it you should update the commit message to reflect it.

**Review by seanbright — CHANGES_REQUESTED** [review]

File: include/asterisk/res_pjsip.h (hunk 1 above)
Comment (pchero) [reply include/asterisk/res_pjsip.h]: Sure, and I believe it's crucial to ensure the commit message accurately reflects the intent of this change. I've updated the message accordingly.

File: include/asterisk/res_pjsip.h (hunk 2 above)
Comment (pchero) [reply include/asterisk/res_pjsip.h]: Np, fixed it.

File: res/ari/internal.h (hunk 3 above)
Comment (pchero) [reply res/ari/internal.h:62]: Sure, and I believe it's crucial to ensure the commit message accurately reflects the intent of this change. I've updated the message accordingly.

**Comment by seanbright** [comment]
Can you squash the 4 commits into just 1?

**Comment by pchero** [comment]
> Can you squash the 4 commits into just 1?

Sure, Np. :)

**Comment by pchero** [comment]
cherry-pick-to: 18
cherry-pick-to: 20
cherry-pick-to: 21
